source/CMakeLists.txt

```diff
@@ -33,6 +33,78 @@ add_clang_executable(binder
   fmt/format.cc
   fmt/posix.cc
   )
+if(USE_EXTERNAL_LLVM)
+find_library(LLVMSO  NAMES LLVM 
+                           LLVM-${LLVM_VERSION_MAJOR} 
+                           LLVM-${LLVM_VERSION_MAJOR}.${LLVM_VERSION_MINOR} 
+                           LLVM-${LLVM_VERSION_MAJOR}.${LLVM_VERSION_MINOR}.${LLVM_VERSION_PATCH}
+                           PATHS  ${LLVM_LIBRARY_DIR}  ${LLVMLIBDIR}
+                           )
+add_definitions(${LLVM_DEFINITIONS})
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${LLVM_COMPILE_FLAGS}")
+list(REMOVE_ITEM CMAKE_CXX_FLAGS "-fno-exceptions")
+string (REPLACE "-fno-exceptions" ""     CMAKE_CXX_FLAGS ${CMAKE_CXX_FLAGS})
+message(STATUS "binder: LLVMSO=${LLVMSO}")
+message(STATUS "binder: CMAKE_CXX_FLAGS=${CMAKE_CXX_FLAGS}")
+if (LLVM_VERSION_MAJOR LESS 7  )
+target_link_libraries(binder PRIVATE
+-Wl,--start-group
+${LLVMSO}
+clang
+clangAnalysis 
+clangARCMigrate
+clangAST
+clangASTMatchers
+clangBasic 
+clangCodeGen
+clangDriver
+clangDynamicASTMatchers
+clangEdit 
+clangFormat 
+clangFrontend
+clangFrontendTool
+clangLex 
+clangParse
+clangRewriteFrontend
+clangSema
+clangSerialization
+clangTooling
+-Wl,--end-group
+)
+endif()
+if (LLVM_VERSION_MAJOR EQUAL 7)
+target_link_libraries(binder
+  PRIVATE
+  ${LLVMSO}
+  clangTooling
+  clangBasic
+  clangASTMatchers
+  clangAST
+  clangFrontend  
```

Comment:
The standalone build uses  a standard add_executable to build executable instead of the llvm machinery.  I suspect the llvm machinery hides some of these libraries.

Reply:
Looked there again.  This if is behind another one : if(USE_EXTERNAL_LLVM), so as for me that makes sense to keep it as is.